src/Riok.Mapperly/Emit/Syntax/SyntaxFactoryHelper.Null.cs

```diff
@@ -33,9 +33,10 @@ public static ExpressionSyntax NullSubstitute(ITypeSymbol t, ExpressionSyntax ar
     {
         return nullFallbackValue switch
         {
+            NullFallbackValue.Default when t.IsNullableValueType() => DefaultExpression(FullyQualifiedIdentifier(t)),
             NullFallbackValue.Default => DefaultLiteral(),
             NullFallbackValue.EmptyString => StringLiteral(string.Empty),
-            NullFallbackValue.CreateInstance => CreateInstance(t),
+            NullFallbackValue.CreateInstance => CreateInstance(t.NonNullable()),
```

Comment:
`CreateInstance` internally always uses a non-nullable identifier, so `NonNullable` is not needed here.

Reply:
This fixed.